### DISCUSSION TITLE
How do i use bullet3 in C project?

### DISCUSSION THREAD

**Discussion by platonvin** [bulletphysics/bullet3#4541] [Q&A]
I have c project that i want to add physics engine to. As far as i see, there is something like C API, but i cant find instructions how to compile for it and how use it. Also, just compiling with mingw gmake on windows (configured with premake4.exe wich says i can choose windows as target OS) fails ("cannot find -lkernel32", is it linux lib?). And i've seen C API repo that is kind of dead and also does not build (undefined references to some functions). There are clearly discussions on forum about C API, and they were defenetly working on it, but feels like it was just removed.
So, how do i use bullet3 in my C project?

**Comment by cuppajoeman** [comment]
Personally I've found success by building this library using cmake I use linux though. What I did to get it working was

```
cmake -S . -B my_build
cd my_build
cmake --build .
```

I agree that it's hard to find instructions on how to get it running as there are various documents but not one centralized truth on the matter, I'm hoping we can make a centralized learning resource in this regard, if you think that's a good idea could you give a :+1: reaction on my issue here? https://github.com/bulletphysics/bullet3/issues/4547
